### DISCUSSION TITLE
OpenGL version for macos

### DISCUSSION THREAD

**Discussion by GZLiew** [Dav1dde/glad#414] [General]
@Dav1dde 

Im not sure is this something I need to use glad to configure

but when I print the GL_VERSION
I got `Version: 2.1 Metal - 83` on my mac m1 arm64

I have set the context to be this

```
  SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
  SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
  SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 3);
```

**Comment by Dav1dde** [comment]
The version you get from `glGetString` is not influenced by glad, so this is purely context (SDL) and driver related. You'll have to see why it only gives you a 2.1 context.

Random code snippet I found [here](https://github.com/MikeShah/SDL2_Tutorials/blob/main/6_OpenGL_SDL2/main.cpp), which is supposed to run on m1:

```c
    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 4);
    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 1);
    SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);

    SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
    SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
```
